std/shared/src/main/scala/cats/effect/std/Hotswap.scala

```diff
@@ -121,7 +121,15 @@ object Hotswap {
       def raise(message: String): F[Unit] =
         F.raiseError[Unit](new RuntimeException(message))
 
-      def shared: Resource[F, Unit] = semaphore.permit
+      def shared(state: Ref[F, State]): Resource[F, Unit] =
+        Resource.makeFull[F, Unit] { poll =>
+          poll(
+            state.get.flatMap {
+              case Acquired(_, _) => semaphore.acquire
+              case _ => F.unit
```

Comment:
We can use a boolean to track whether the semaphore should be released :)

```suggestion
              case Acquired(_, _) => semaphore.acquire.as(true)
              case _ => F.pure(false)
```

Reply:
Oh, also in this case, the scope of the `poll` should be restricted to `semaphore.acquire`

Reply:
Done, thank you!